src/lib/markbind/src/parsers/componentParser.js

```diff
@@ -119,6 +119,41 @@ function _assignPanelId(element) {
   }
 }
 
+/**
+ * Check and warns if element has conflicting attributes.
+ * @param element Root element to check
+ * @param attribute An attribute that is conflicting with other attributes
+ * @param attrsConflictingWith The attributes conflicting with `attribute`
+ */
+function _warnConflictingAttributes(element, attribute, attrsConflictingWith) {
+  if (!(attribute in element.attribs)) {
+    return;
+  }
+  for (let i = 0; i < attrsConflictingWith.length; i += 1) {
+    if (attrsConflictingWith[i] in element.attribs) {
+      // TODO: Use logger here instead of console.warn. See issue #1060
+      // eslint-disable-next-line no-console
+      console.warn(`warn: Usage of conflicting ${element.name} attributes: `
+        + `'${attribute}' with '${attrsConflictingWith[i]}'`);
+    }
+  }
+}
+
+/**
+ * Check and warns if element has a deprecated attribute.
+ * @param element Root element to check
+ * @param attribute An attribute that is deprecated.
+ * @param attrsInconsistentWith The attribute to be used instead
+ */
+function _warnDeprecatedAttribute(element, deprecatedAttrib, correctAttrib) {
```

Comment:
What do you think of taking as input an object of `{deprecated: correct}` instead? This will allow us to handle multiple deprecated attributes in one function instead of having to call this multiple times.

Reply:
Sorry, I don't understand. 

`_warnDeprecatedAttribute(element, 'title', 'header');`
becomes
`_warnDeprecatedAttribute(element, {'title', 'header'});` ?
You cannot have a string as a object key. Unless you mean `title` will be the name of the key itself?

How does a method call of multiple deprecated attributes look like for your case?

Reply:
Consider we have: the attribute `title` to be replaced with `header` and the attribute `some-other-key` to be replaced with `corrected-key`.

I imagine the function call would look something like this:

`_warnDeprecatedAttribute(element, {title: 'header', 'some-other-key': 'corrected-key'});`

https://stackoverflow.com/a/24243895 might give a little more context about why object keys are strings.